app/models/order.rb

```diff
@@ -16,50 +31,12 @@ def complete_building
   include Submission::AccessionBehaviour
   include ModelExtensions::Order
 
-  class CompositeAttribute
-    attr_reader :display_name, :key, :default, :options
-    def initialize(key)
-      @key = key
-    end
-
-    def add(attribute, metadata)
-      @display_name ||= attribute.display_name
-      @key            = attribute.assignable_attribute_name
-      @default      ||= attribute.find_default(nil, metadata)
-      @kind           = attribute.kind if @kind.nil? || attribute.required?
-      if attribute.selection?
-        new_options   = attribute.selection_options(metadata)
-        @options    ||= new_options if selection?
-        @options     &= new_options
-      end
-    end
-
-    def kind
-      @kind || FieldInfo::TEXT
-    end
-
-    def selection?
-      kind == FieldInfo::SELECTION
-    end
-
-    def to_field_infos
-      values = {
-        display_name: display_name,
-        key: key,
-        default_value: default,
-        kind: kind
-      }
-      values.update(selection: options) if selection?
-      FieldInfo.new(values)
-    end
-  end
-
```

Comment:
This behaviour has been integrated in field infos to allow them to be combined with &